client/lib/sentry/index.ts

```diff
@@ -132,13 +132,11 @@ export async function initSentry( { beforeSend, userId }: SentryOptions ) {
 		}
 		state = { state: 'loading' };
 
-		// Enable Sentry only for 10% of requests or always in calypso.live for testing.
-		// Always disable if catch-js-errors is not available in the environment.
+		// Enable Sentry only for 10% of requests. Disable if catch-js-errors is not available in the environment.
+		// When always-enable-sentry is available, bypass the other checks.
 		if (
-			! (
-				config.isEnabled( 'catch-js-errors' ) &&
-				( config( 'env_id' ) === 'wpcalypso' || Math.floor( Math.random() * 10 ) === 1 )
-			)
+			! config.isEnabled( 'always-enable-sentry' ) &&
```

Comment:
Not a blocker at all, but I just an outburst about conditions with multiple negated expression -- they tend to trick me a bit, I have to read a couple of times to "evaluate" the and understand it... on the other hand, the following reads more like plain english to me:

config.isEnabled( 'always-enable-sentry' ) || 
			( config.isEnabled( 'catch-js-errors' ) && Math.floor( Math.random() * 10 ) == 1 ) )

I'd also put the `Math.floor...` in a `in10thPercentileSegment()` or something. Food for thought for a future improvement if you think it'd be worth it; I can understand if that's also subjective and a matter of personal preference.

Reply:
True, this is a weird one. I like your revision, but we're basically checking if we need to disable it, not to enable it. I opened #79621 to improve it! I ended up splitting the entire check into another function